pkg/systemd/overview-cards/cryptoPolicies.jsx

```diff
@@ -0,0 +1,199 @@
+    LEGACY: _("Provides settings for ensuring maximum compatibility with legacy systems. This policy is less secure and intended to be a easy way to switch system to be compatible with older systems. It should provide at least 80-bit security and excludes 3DES and RC4."),
+};
+const defaultpolicy = "DEFAULT";
+const stampfile = "/run/cockpit/crypto-policies-reboot-stamp";
+
+export const CryptoPolicyRow = () => {
+    const [showCryptoPolicy, setShowCryptoPolicy] = useState(false);
+    const [currentCryptoPolicy, setCurrentCryptoPolicy] = useState("");
+    const [isOpen, setIsOpen] = useState();
+
+    useEffect(() => {
+        // Avoid cockpit.spawn every re-render from parent
+        if (!showCryptoPolicy) {
+            cockpit.spawn(["update-crypto-policies", "--show"]).then(output => {
+                setCurrentCryptoPolicy(output.trim());
+                setShowCryptoPolicy(true);
+            });
+        }
+    }, [showCryptoPolicy]);
+
+    if (!showCryptoPolicy) {
+        return null;
+    }
+    const permission = cockpit.permission({ admin: true });
+
+    return (
+        <tr>
+            <th scope="row">{_("Crypto policy")}</th>
+            <td>
+                <PrivilegedButton variant="link" buttonId="crypto-policy-button" tooltipId="tip-cpu-security"
+                                excuse={ _("The user $0 is not permitted to change crypto policies") }
+                                permission={ permission } onClick={() => setIsOpen(true)}>
+                    {currentCryptoPolicy}
+                </PrivilegedButton>
+                {isOpen &&
+                <CryptoPolicyDialog close={() => setIsOpen(false)}
+                          currentCryptoPolicy={currentCryptoPolicy}
+                          setCurrentCryptoPolicy={setCurrentCryptoPolicy}
+                />
+                }
+            </td>
+        </tr>
+    );
+};
+
+const CryptoPolicyDialog = ({
+    close,
+    currentCryptoPolicy,
+    setCurrentCryptoPolicy,
+}) => {
+    const [error, setError] = useState();
+    const [selected, setSelected] = useState();
+
+    const setPolicy = (reboot) => {
+        cockpit.spawn(["update-crypto-policies", "--set", selected], { superuser: "require", err: "message" }).then(() => {
+            setCurrentCryptoPolicy(selected);
+            if (reboot) {
+                cockpit.spawn(["shutdown", "--reboot", "now"], { superuser: "require", err: "message" })
+                        .catch(error => setError(error));
+            } else {
+                cockpit.spawn(["touch", stampfile], { superuser: "require", err: "message" }).then(() => {
+                    close();
+                });
+            }
+        })
+                .catch(error => {
```

Comment:
This is either mis-indented or misplaced.

Reply:
Hmm, it's something eslint seems to format for me.

Reply:
Oh, I see -- please move the cockpit.spawn()'s `.then` on the next line, then it's properly aligned.